sktime/datatypes/_convert.py

```diff
@@ -66,12 +65,17 @@
     "mtype",
 ]
 
+import numpy as np
+import pandas as pd
 
 from sktime.datatypes._series import convert_dict_Series
 from sktime.datatypes._series import infer_mtype_dict_Series
 
 from sktime.datatypes._panel import convert_dict_Panel
```

Comment:
Do we not have `infer_mtype_dict_Panel` because there is only one mtype for Panel data?

Reply:
Actually, I'm looking to deprecate the `infer_mtype_dict`-s, because there is no 1:1 mapping onto types.
Perhaps I should have simply deleted this.

Instead, it will be handled by the `mtype` function which loops through the checks.
In all the checks (in `check_is`), the first line is a type check, so the "first level of checks" is equivalent to using the dictionary.

Reply:
more precisely, there are two `mtype` functions: the old one, in `_convert`, and a new one, in `_check`.
I want to deprecate the one in `_convert` (currently only does it for `Series`) and replace it by the one in `_check` (this one is generic and relies on full checks rather than dictionaries).

Reply:
might be worth adding a todo item or comment saying that the dicts are to be deprecated/deleted

Reply:
I decided to removed the old `mtyp` function and replaced it by the new one in the module export.

Since everything was referring to the module export rather than the precise location, and both functions had the same signature, nothing broke - so no deprecation is actually needed, and the problem should be fixed?